flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/writer/FileWriterBucketStateSerializer.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.file.sink.writer;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.core.io.SimpleVersionedSerialization;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+import org.apache.flink.core.memory.DataInputDeserializer;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputSerializer;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+
+import static org.apache.flink.streaming.api.functions.sink.filesystem.InProgressFileWriter.InProgressFileRecoverable;
+
+/**
+ * A {@code SimpleVersionedSerializer} used to serialize the {@link FileWriterBucketState BucketState}.
+ */
+@Internal
+public class FileWriterBucketStateSerializer<BucketID>
+		implements SimpleVersionedSerializer<FileWriterBucketState<BucketID>> {
+
+	private static final int MAGIC_NUMBER = 0x1e764b79;
+
+	private final SimpleVersionedSerializer<InProgressFileRecoverable> inProgressFileRecoverableSerializer;
+
+	private final SimpleVersionedSerializer<BucketID> bucketIdSerializer;
+
+	public FileWriterBucketStateSerializer(
+			SimpleVersionedSerializer<InProgressFileRecoverable> inProgressFileRecoverableSerializer,
+			SimpleVersionedSerializer<BucketID> bucketIdSerializer) {
+		this.inProgressFileRecoverableSerializer = Preconditions.checkNotNull(
```

Comment:
Why not statically importing the `checkNotNull()` check :
```
import static org.apache.flink.util.Preconditions.checkNotNull;
```
so that you do not have to write the whole `Preconditions.checkNotNull()` ?